pysr/sr.py

```diff
@@ -1394,48 +1334,318 @@ def _run(self, X, y, weights, variable_names):
+        cprocs = 0 if self.multithreading else self.procs
 
         # Call to Julia backend.
         # See https://github.com/search?q=%22function+EquationSearch%22+repo%3AMilesCranmer%2FSymbolicRegression.jl+path%3A%2Fsrc%2F+filename%3ASymbolicRegression.jl+language%3AJulia&type=Code
-        self.raw_julia_state = Main.EquationSearch(
+        self.raw_julia_state_ = Main.EquationSearch(
             Main.X,
             Main.y,
             weights=Main.weights,
-            niterations=int(self.params["niterations"]),
-            varMap=(
-                variable_names
-                if selection is None
-                else [variable_names[i] for i in selection]
-            ),
+            niterations=int(self.niterations),
+            varMap=self.feature_names_in_.tolist(),
             options=options,
             numprocs=int(cprocs),
-            multithreading=bool(multithreading),
-            saved_state=self.raw_julia_state,
-            addprocs_function=cluster_manager,
+            multithreading=bool(self.multithreading),
+            saved_state=self.raw_julia_state_,
+            addprocs_function=self.cluster_manager,
         )
 
-        self.variable_names = variable_names
-        self.selection = selection
+        # Set attributes
+        self.equations_ = self.get_hof()
 
-        # Not in params:
-        # selection, variable_names, multioutput
+        if self.delete_tempfiles:
+            shutil.rmtree(self.tempdir_)
 
-        self.equations = self.get_hof()
+        already_ran = True
 
-        if self.params["delete_tempfiles"]:
-            shutil.rmtree(tmpdir)
+        return self
 
-        already_ran = True
+    def fit(
+        self,
+        X,
+        y,
+        Xresampled=None,
+        weights=None,
+        variable_names=None,
+        from_equation_file=False,
+    ):
+        """
+        Search for equations to fit the dataset and store them in `self.equations_`.
+
+        Parameters
+        ----------
+        X : {ndarray | pandas.DataFrame} of shape (n_samples, n_features)
+            Training data.
+
+        y : {ndarray | pandas.DataFrame} of shape (n_samples,) or (n_samples, n_targets)
+            Target values. Will be cast to X's dtype if necessary.
+
+        Xresampled : {ndarray | pandas.DataFrame} of shape
+                        (n_resampled, n_features), default=None
+            Resampled training data used for denoising.
+
+        weights : {ndarray | pandas.DataFrame} of the same shape as y, default=None
+            Each element is how to weight the mean-square-error loss
+            for that particular element of y.
+
+        variable_names : list[str], default=None
+            A list of names for the variables, other than "x0", "x1", etc.
+            If :param`X` is a pandas dataframe, the column names will be used.
+            If variable_names are specified
+
+        from_equation_file : bool, default=False
```

Comment:
If checkpoints can only be loaded for fitted estimators, creating a new `load_from_checkpoint` or `load` function would be no issue.  Something like this would probably work nicely: 

However for unfitted estimators it is quite difficult as `get_hof` needs to know fit dependant information such as `selection_mask_`, `feature_names_in_`, and `nout_`. The current implmentation with `from_equation_file` was the simplest way around this I could think of. If feature selection didn't occur as part of the fit (feature selection was performed outside of the PySRRegressor), then `selection_mask_` is not needed and `feature_names_in_` could be overwritten by a parameter in a new `model.load_from_checkpoint` method.

Reply:
This sounds good to me!